admin/spec/service/pupil-census.service.spec.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+/* global spyOn, describe, it, expect, fail */
+
+const pupilCensusService = require('../../services/pupil-census.service')
+const pupilCensusDataService = require('../../services/data-access/pupil-census.data.service')
+const azureFileDataService = require('../../services/data-access/azure-file.data.service')
+
+const pupilCensusMock = {
+  'uuid': 'bfa9ab1b-88ae-46f2-a4ff-726c0567e37c',
+  'field': 'csvPupilCensusFile',
+  'file': 'spec/files/Pupil-Census-Data.csv',
+  'filename': 'Pupil-Census-Data.csv',
+  'encoding': '7bit',
+  'mimetype': 'text/csv',
+  'truncated': false,
+  'done': true
+}
+
+describe('pupilCensusService', () => {
+  describe('upload', () => {
+    it('calls uploadToBlobStorage when reading is done', async () => {
+      spyOn(pupilCensusService, 'uploadToBlobStorage')
+      spyOn(pupilCensusDataService, 'sqlCreate')
+      await pupilCensusService.upload(pupilCensusMock)
+      expect(pupilCensusService.uploadToBlobStorage).toHaveBeenCalled()
+    })
+    xit('rejects if uploadToBlobStorage fails', async () => {
```

Comment:
Why is this marked as pending with `xit`?

Reply:
@danwilliams it was throwing a promise rejection in the console when running the test suite even though the test was passing.

Reason is I was capturing the rejection in the event queue at a later stage and not in the actual call stack where the promise is created.

That has been updated now